File: ios/Tiktok.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Tiktok, NSObject)
  RCT_EXTERN_METHOD(
                    auth: (NSArray <NSString *>*)scopes
                    callback: (RCTResponseSenderBlock)callback
                    )
@end

File: ios/Tiktok.swift
import Foundation
import TikTokOpenSDK
import Photos

@objc(Tiktok)
class Tiktok: UIViewController {

  @objc
  func auth(_ scopes: [String], callback: @escaping RCTResponseSenderBlock) {
    let scopesSet = NSOrderedSet(array:scopes)
    let request = TikTokOpenSDKAuthRequest()
    request.permissions = scopesSet

    DispatchQueue.main.async {
      request.send(self, completion: { resp -> Void in
        callback([
          ["status": resp.errCode.rawValue, "code": resp.code]
        ])
      })
    }
  }
}
